netdata-installer.sh

```diff
@@ -862,28 +795,23 @@ install_netdata_service || run_failed "Cannot install netdata init service."
 # -----------------------------------------------------------------------------
 # check if we can re-start netdata
 
+# TODO(paulfantom): Creation of configuration file should be handled by a build system. Additionally we shouldn't touch configuration files in /etc/netdata/...
 started=0
 if [ ${DONOTSTART} -eq 1 ]; then
-	generate_netdata_conf "${NETDATA_USER}" "${NETDATA_PREFIX}/etc/netdata/netdata.conf" "http://localhost:${NETDATA_PORT}/netdata.conf"
-
+	create_netdata_conf "${NETDATA_PREFIX}/etc/netdata/netdata.conf"
 else
-	restart_netdata ${NETDATA_PREFIX}/usr/sbin/netdata "${@}"
-	if [ $? -ne 0 ]; then
-		echo >&2
-		echo >&2 "SORRY! FAILED TO START NETDATA!"
-		echo >&2
-		exit 1
+	if ! restart_netdata "${NETDATA_PREFIX}/usr/sbin/netdata" "${@}"; then
+		fatal "Cannot start netdata!"
 	fi
 
 	started=1
-	echo >&2 "OK. NetData Started!"
-	echo >&2
-
-	# -----------------------------------------------------------------------------
-	# save a config file, if it is not already there
-
-	download_netdata_conf "${NETDATA_USER}" "${NETDATA_PREFIX}/etc/netdata/netdata.conf" "http://localhost:${NETDATA_PORT}/netdata.conf"
+	run_ok "netdata started!"
+	create_netdata_conf "${NETDATA_PREFIX}/etc/netdata/netdata.conf" "http://localhost:${NETDATA_PORT}/netdata.conf"
```

Comment:
should this be `download_netdata_conf`?

Reply:
No as this PR removes `download_netdata_conf`.

`download_netdata_conf`, `fix_netdata_conf`, and `generate_netdata_conf` are merged into one `create_netdata_conf` function.